src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/HyperlinkClickedTelemetryEvent.cs

```diff
@@ -22,10 +47,74 @@ public HyperlinkClickedTelemetryEvent(HyperlinkType hyperlinkType, ContractsItem
             base[IsSolutionViewPropertyName] = isSolutionView;
         }
 
-        public HyperlinkClickedTelemetryEvent(HyperlinkType hyperlinkType, ContractsItemFilter currentTab, bool isSolutionView, string alternativePackageId)
+        public HyperlinkClickedTelemetryEvent(HyperlinkType hyperlinkType, ContractsItemFilter currentTab, bool isSolutionView, string alternativePackageId, PackageSourceMoniker selectedSource)
             : this(hyperlinkType, currentTab, isSolutionView)
         {
+            if (selectedSource == null)
+            {
+                throw new ArgumentNullException(nameof(selectedSource));
+            }
+
+            var feedInfo = GetFeedInfo(selectedSource);
             AddPiiData(AlternativePackageIdPropertyName, VSTelemetryServiceUtility.NormalizePackageId(alternativePackageId));
+            SelectedSourceHost = feedInfo.Item3;
+            SelectedSourceHostNameType = feedInfo.Item2;
+            SelectedSourceType = feedInfo.Item1;
+            HasUnknownRemoteFeed = ContainsUnknownRemoteFeed(selectedSource);
+        }
+
+        internal static bool ContainsUnknownRemoteFeed(PackageSourceMoniker selectedSource)
```

Comment:
Can this be better named and/or commented? Is the point to filter out HTTP (not HTTPS) feeds?